### PR TITLE
le benchmark ne fonctionne plus

### PR DESCRIPTION
**Pull request by theophilestourbe**
Bonjour monsieur, 
Depuis que nous avons fork votre repo nous avons des problèmes avec le benchmark, voici l'erreur:
image
Ça dit que je n'ai pas dotnet d'installé alors que j'ai pourtant bien le sdk .NET v8.0.203 sur un environment Rider, de plus je ne rencontrais pas cette erreur avant lorsque je le lançais. Le single solver test marche toujours

### CODE REVIEW

**Comment by jsboigeEpita** [comment]
Bonjour,

J'ai merged votre solution et ai pu vérifier que vous avez des modèles qui fonctionnent même si j'ai des problèmes de chemin sur les modèles tensorflow.
Puisque vous avez pu archiver vos modèles préentraînés dans la solution, l'idéal serait de les inclure comme ressources partagées au même titre que les scripts, et de les déployer à la demande en testant si les fichiers existent.
Le problème avec les chemins relatifs, est que Benchmark.Net a sont propre fonctionnement où il crée des sous-répertoires, du coup, vous n'êtes pas sûr du répertoire d'exécution qui va changer entre le test simple d'un solver et l'exécution dans un Benchmark.

Concernant les benchmarks, j'ai effectué des corrections précédemment qui me paraissaient correctes car elles semblaient avoir résolu un problème d'exécution signalé dans la classe. Je crois bien que mes corrections précédentes n'étaient pas les bonnes, et j'ai fini par trouver aujourd'hui que le problème était lié au Sudoku trivial de "warmup" d'itération, qui était exécuté sans limite de temps.
De mon côté, tout tourne bien, mais si vous avez des soucis du vôtre, vous pouvez sans doute jeter un oeil dans BenchmarkSolvers.cs et comparer les paramètres de Config actuels avec la version d'avant. Pas de souci de mon côté pour les reprendre de sorte que ça puisse fonctionner chez tout le monde, du fait que je pense avoir corrigé le vrai problème autrement. 

Enfin, je vois que effectuez systématiquement une résolution en 1 seule passe, et j'ai l'impression qu'en l'état, vos modèles ne parviennent pas à résoudre les sudokus difficiles. Il me semblent que d'autres élèves sont arrivés à résoudre des Sudokus plus difficiles avec des modèles moins performants, en utilisant le trick de ne garder que les softmax les plus probant et en réinjectant les puzzle partiellement résolus pour les autres.
